src/File.php

```diff
@@ -390,27 +368,50 @@ public function get_body_content(): string
         return (string) $this->body;
     }
 
-    private function get_from_headers(string $name): array
+    /**
+     * Check if the $headers property was changed and update the internal state accordingly.
+     */
+    private function maybe_update_headers(): void
```

Comment:
I'm not sure why we need this? Why not just use the previous `get_from_headers()` method?

Reply:
Thank you for the explanation and for your attention to this detail.